app/questionnaire/placeholder_parser.py

```diff
@@ -193,3 +206,22 @@ def _get_routing_path_block_ids(
     def _all_value_sources_metadata(self, placeholder: Mapping) -> bool:
         sources = self._schema.get_values_for_key(placeholder, key="source")
         return all(source == "metadata" for source in sources)
+
+    def _get_value_source_resolver_for_transform(
+        self, transform_name: str
+    ) -> ValueSourceResolver:
+        if not (
+            section_ids := self._schema.get_placeholder_dependencies(transform_name)
+        ):
+            return self._value_source_resolver
+
+        complete_routing_path_block_ids = []
+
+        for section_id in section_ids:
+            routing_path_block_ids = self._path_finder.routing_path(
+                section_id=section_id  # type: ignore
+                # section_id always exists
+            ).block_ids
```

Comment:
1. You need to use the `self._routing_path_block_ids_by_section_key` cache, as the routing path existing alredy.
2. This doesn't take into account repeating sections, so you need to handle that.

Reply:
Forgot to reply to this comment. Way we get the value source resolver is now different and should follow how we did it with calculated summaries. Also now supports repeating sections.